src/lib/isISBN.js

```diff
@@ -1,43 +1,55 @@
 import assertString from './util/assertString';
 
-const isbn10Maybe = /^(?:[0-9]{9}X|[0-9]{10})$/;
-const isbn13Maybe = /^(?:[0-9]{13})$/;
+const possibleIsbn10 = /^(?:[0-9]{9}X|[0-9]{10})$/;
+const possibleIsbn13 = /^(?:[0-9]{13})$/;
 const factor = [1, 3];
 
-export default function isISBN(str, version = '') {
-  assertString(str);
-  version = String(version);
-  if (!version) {
-    return isISBN(str, 10) || isISBN(str, 13);
+export default function isISBN(isbn, options) {
+  assertString(isbn);
+
+  // For backwards compatibility:
+  // isISBN(str [, version]), i.e. `options` could be used as argument for the legacy `version`
+  const version = String(options?.version || options);
+
+  if (!(options?.version || options)) {
+    return isISBN(isbn, { version: 10 }) || isISBN(isbn, { version: 13 });
   }
-  const sanitized = str.replace(/[\s-]+/g, '');
+
+  const sanitizedIsbn = isbn.replace(/[\s-]+/g, '');
+
   let checksum = 0;
-  let i;
+
   if (version === '10') {
-    if (!isbn10Maybe.test(sanitized)) {
+    if (!possibleIsbn10.test(sanitizedIsbn)) {
       return false;
     }
-    for (i = 0; i < 9; i++) {
-      checksum += (i + 1) * sanitized.charAt(i);
+
+    for (let i = 0; i < version - 1; i++) {
+      checksum += (i + 1) * sanitizedIsbn.charAt(i);
     }
-    if (sanitized.charAt(9) === 'X') {
+
+    if (sanitizedIsbn.charAt(9) === 'X') {
       checksum += 10 * 10;
     } else {
-      checksum += 10 * sanitized.charAt(9);
+      checksum += 10 * sanitizedIsbn.charAt(9);
     }
+
     if ((checksum % 11) === 0) {
-      return !!sanitized;
+      return !!sanitizedIsbn;
```

Comment:
Is there a reason, why we can't just put `true` here?
That would be a bit easier to read to me, and what I can see here that will always be the output of `!!sanitizedIsbn` anyways (unless it is an empty string - which line 23 already takes care of anyway).
(same thing applies to line 50 as well)
correct me if I am wrong though!

Reply:
I think that should work, will test and update it

Reply:
Works fine. Thanks for the suggestion!